packages/react/select/src/Select.tsx

```diff
@@ -624,7 +625,25 @@ const SelectContentImpl = React.forwardRef<SelectContentImplElement, SelectConte
       }
     });
 
+    const handleItemSeen = React.useCallback(
+      (node: SelectItemElement | null, value: string, disabled: boolean) => {
+        const isFirstSeenValidItem = !firstItemRef.current && !disabled;
+        const isSelectedItem = context.value !== undefined && context.value === value;
+        if (isSelectedItem || isFirstSeenValidItem) {
+          setSelectedItem(node);
+          if (isFirstSeenValidItem) firstItemRef.current = true;
+        }
+      },
+      [context.value]
+    );
     const handleItemLeave = React.useCallback(() => content?.focus(), [content]);
+    const handleItemTextSeen = React.useCallback(
+      (node: SelectItemTextElement | null, value: string) => {
+        const isSelectedItemText = context.value !== undefined && context.value === value;
+        if (isSelectedItemText || !firstItemRef?.current) setSelectedItemText(node);
```

Comment:
`!firstItemRef?.current` feels incidental based on React rendering, will the if block run for every iteration up until the parent item passes the validity check? would that be expected?

Reply:
I don't think that is incidental, because React always renders the children first. So it will let any item text through until it finds its parent item which is the first non-disabled one.

Reply:
If I have a list of 4 items and the first 3 are disabled, `setSelectedItemText` be called incorrectly 3 times followed by a valid fourth, this is in contrast to `setSelectedItem` which is only called once.  

Is that what we want? shouldn't the behaviour match?

Reply:
oh you mean specifically because of the `disabled` check? Yeah I guess we could add a disabled check in the item text too to make it equivalent if that's what you mean?

Reply:
Yes, sorry, in effect that’s what I meant. My bigger point was that the code expects the parent `Item` to run before `ItemText`, this doesn't happen because React renders children first, and so one behaves differently until the flag is updated by the parent.